File: algorithms/2006.count-number-of-pairs-with-absolute-difference-k/solution.rs
impl Solution {
    pub fn count_k_difference(nums: Vec<i32>, k: i32) -> i32 {
        let mut count = 0i32;

        for i in 0..(nums.len() - 1) {
            for j in (i + 1)..nums.len() {
                if (nums[i] - nums[j]).abs() == k {
                    count += 1;
                }
            }
        }

        count
    }
}

File: algorithms/2006.count-number-of-pairs-with-absolute-difference-k/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        int count = 0;
        size_t size = nums.size();

        for (int i = 0; i < (size - 1); i++) {
            for (int j = (i + 1); j < size; j++) {
                if (abs(nums[i] - nums[j]) == k) {
                    count++;
                }
            }
        }

        return count;
    }
};
